### PR TITLE
Validation added

### PR DESCRIPTION
**Pull request by RainiWijewantha**
Only Job Post Module

### CODE REVIEW

File: src/main/java/com/hris/HRIS/service/JobPostService.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+@Service
+public class JobPostService {
+
+    @Autowired
+    JobPostRepository jobPostRepository;
+
+    //validate email
+    public boolean isValidEmail(String email) {
+        String emailRegex = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$";
```
Comment (kavicastelo) [inline src/main/java/com/hris/HRIS/service/JobPostService.java:17]: Using regex for validation is a bit of an advanced and I think you tested this pattern before opening the PR. Regex is are bit complicated to handle in different scenarios. As an example, the email validation pattern is different in Java and C or Java and Javascript.
`^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$`
To my knowledge, you allowed simple and capital letters, 0-9 numbers, and a couple of special characters (._%+-) for the first element. We know commonly we need to add `\` before special characters we going to accept. Then you'll check the email containing the `@` symbol. Then you try to allow all English characters and dot and hyphen symbols for the next element (gmail, yahoo). then you checking the dot is contained or not. After that, I see you allowing the final element to be only 2 characters. How do you allow (com) emails using 2 chars?

**Review by kavicastelo — CHANGES_REQUESTED** [review]
Provide answers and change requested changes. If you are facing any problems with that, contact me.

**Comment by RainiWijewantha** [comment]
Ok, I'll change

**Comment by kavicastelo** [comment]
You're spending too much time to complete your requested changes. Your PR marked as red because your PR not made any changes within 4 business days. Hurry up!!
@RainiWijewantha

**Pull request closed without merging**